files/class-api-client.php

```diff
@@ -152,13 +152,20 @@ private function calculate_upload_timeout( $file_size ) {
 
 	public function get_file( $file_path ) {
 		// check in cache first
-		$file_content = $this->cache->get_file( $file_path );
-		if ( $file_content ) {
-			return $file_content;
+		$file = $this->cache->get_file( $file_path );
+		if ( $file ) {
+			return $file;
 		}
 
+		$tmp_file = tempnam( get_temp_dir(), 'vip' );
+
+		$request_args = [
+			'stream' => true,
+			'filename' => $tmp_file,
+		];
```

Comment:
What do these args do in `$this->call_api()`? Looks like they are passed through to the files service - a comment explaining what it's doing would be helpful.

Reply:
It is ultimately used as arguments for `wp_remote_request()` but I agree, some comments would help clear that up.

Reply:
Does it mean that `wp_remote_request()` will stream the response to the file?

Reply:
Yes. If `stream` is `true` then `wp_remote_request()` will stream the response into `filename`